### DISCUSSION TITLE
rpi zero 2 w

### DISCUSSION THREAD

**Discussion by asutrem** [geerlingguy/pi-webcam#50] [Ideas]
Hi Jeff!

Would this project work on the new rpi zero 2? Was thinking that the new processor and better image processing might help the zero finally be able to put out 1080p video without dropping frames. 

Thanks!

**Comment by geerlingguy** [comment]
I think it _should_, though I haven't tested it yet.

**Reply by asutrem** [reply]
Ok! I guess I'll try it out when I get my hands on the zero 2. Might take a while to get stock in my area but will see how it goes.

**Comment by acidthefurry** [comment]
hello there, i personally HAVE tried and it works great,
also i would advice using raspbian lite instead of the desktop one, my camera now boots in ~14 seconds to usable

**Reply by acidthefurry** [reply]
personally i have it running at 1080p60, and havent seen a "green" frame ever since i upgraded to the 2,
(though before it where singular frames, and where on an ignorable amount)

**Reply by jsimpsonUoA** [reply]
Hi there, could you please let me know the exact image you used to get this working on the Pi Zero 2W? I've tried with the latest desktop version (2021-10-30-raspios-bullseye-armhf), but haven't been able to get it working.

**Reply by acidthefurry** [reply]
@jsimpsonUoA i used raspbian buster lite and upgraded to bullseye

**Reply by jsimpsonUoA** [reply]
Ok, thanks. I've got it working now with the Zero 2W using 2021-05-07-raspios-buster-armhf-lite.

For reference, I tried using both buster and bullseye desktop versions, but neither worked. Buster hung on the boot screen. I seems like you may have to use the Lite version for now??

**Reply by acidthefurry** [reply]
i mean lite boots way faster anyways

**Comment by qlkv** [comment]
I have tried different builds of OS and followed the instructions — but nothing works on Raspberry Pi Zero 2W, the device simply doesn't show up in the settings when I plug in to my computer. I decided to give up and just buy a ready-made webcam.

**Comment by dquadros** [comment]
After a few bad tries, I got it **almost** working. Here are the steps I took, **highlighting** the differences to the steps at the README.md:

1) Use the Raspberry Pi Imager to write the **"Raspberry Pi OS (Legacy, 32-bit) Lite"** image (dated 2024-10-22) to the SD. You can configure WiFi and SSH in the Imager, but **make sure the user name is pi**
2) **In the first boot, run sudo raspi=config and enable legacy camera support at Interface Options. Reboot (sudo reboot)**
3) sudo apt update
4) sudo apt install -y python3-dev python3-pip libyaml-dev libffi-dev git
5) sudo pip3 install --no-binary pyyaml ansible  (this takes some time to run)
6) git clone https://github.com/geerlingguy/pi-webcam.git
7) cd pi-webcam
8) cp inventory-local.example inventory
9) ansible-playbook main.yml
10) sudo poweroff

Installation goes ok and the camera is recognized by Windows 10. Alas, it stops working after less than a second sending video to the Camera application.
